modules/lightmapper_rd/lightmapper_rd.cpp

```diff
@@ -1501,8 +1632,24 @@ LightmapperRD::BakeError LightmapperRD::bake(BakeQuality p_quality, bool p_use_d
 		}
 
 		{
-			SWAP(light_accum_tex, light_accum_tex2);
-			BakeError error = _denoise(rd, compute_shader, compute_base_uniform_set, push_constant, light_accum_tex2, normal_tex, light_accum_tex, p_denoiser_strength, atlas_size, atlas_slices, p_bake_sh, p_step_function);
+			String oidn_path = EDITOR_GET("filesystem/tools/oidn/oidn_path");
+			Ref<DirAccess> da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+
+			if (da->dir_exists(oidn_path)) {
+				if (OS::get_singleton()->get_name() == "Windows") {
+					oidn_path = oidn_path.path_join("oidnDenoise.exe");
+				} else {
+					oidn_path = oidn_path.path_join("oidnDenoise");
+				}
+			}
+
+			BakeError error;
+			if (!oidn_path.is_empty() && da->file_exists(oidn_path)) {
+				error = _denoise_oidn(rd, light_accum_tex, normal_tex, light_accum_tex, atlas_size, atlas_slices, p_bake_sh, oidn_path);
+			} else {
```

Comment:
We would need to expose more configuration to decide what denoiser should be used, I don't think it should rely solely on whether the OIDN path was configured. CC @DarioSamo

Reply:
I agree with that, it should probably be an option (lightmapper or project settings?) and just throw an error message if OIDN is missing when it's attempted to be used to indicate to the user they should download it and set the path on the project settings.

Reply:
Added `rendering/lightmapping/denoiser` project setting to select denoiser.